functional/test/test_streams.py

```diff
@@ -145,3 +147,40 @@ def test_to_csv(self):
         sequence.to_csv(tmp_path)
         result = seq.csv(tmp_path).to_list()
         self.assertEqual(expect, result)
+
+    def test_to_sqlite3_failure(self):
+        # test failure case
+        insert_sql = "INSERT INTO user (id, name) VALUES (?, ?)"
+        elements = [(1, "Tom"), (2, "Jack"), (3, "Jane"), (4, "Stephan")]
+        with self.assertRaises(ValueError):
+            seq(elements).to_sqlite3(1, insert_sql)
+
+    def test_to_sqlite3_file(self):
+        tmp_path = 'functional/test/data/tmp/{}.db'.format(time.time())
+
+        with sqlite3.connect(tmp_path) as conn:
+            conn.execute("CREATE TABLE user (id INT, name TEXT);")
+            conn.commit()
+
+        insert_sql = "INSERT INTO user (id, name) VALUES (?, ?)"
+        elements = [(1, "Tom"), (2, "Jack"), (3, "Jane"), (4, "Stephan")]
+
+        # test insert into a file
+        seq(elements).to_sqlite3(tmp_path, insert_sql)
+        result = seq.sqlite3(tmp_path, "SELECT id, name FROM user;").to_list()
+        self.assertListEqual(elements, result)
+
+        os.remove(tmp_path)
```

Comment:
Since this isn't handled anywhere else (like `run-test.sh`), should probably encapsulate everything in a try/finally block. In the finally block should do the remove and also catch `FileNotFoundError` in case its already been deleted.